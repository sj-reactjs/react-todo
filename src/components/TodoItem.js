import React from 'react'

function TodoItem(props) {
    const {
        id,
        task,
        description,
        status,
        created_on,
        last_updated_on,
        remove,
        setStage
    } = props

    return (
        <div role="listitem" className="todo-card">
            <div className="card-actions">
                <div role="button" tabIndex="0" data-action="edit" onClick={() => setStage('edit', id)}>Edit</div>
                <div className="created">
                    <i className="fa fa-clock-o"></i>{created_on !== '' ? created_on : 'N/A'}
                </div>
                <div role="button" tabIndex="0" className="right close" onClick={() => remove(id)}>
                    <i className={`fa fa-trash`}></i>
                </div>
            </div>
            <div className="title">
                <h3>{task}</h3>
            </div>
            <div className="description">
                <p>{description}</p>
            </div>
            <div className="status">{status}</div>
            <div className="updated">{last_updated_on !== '' ? last_updated_on : 'N/A'}</div>
        </div>
    )
}

export { TodoItem }
