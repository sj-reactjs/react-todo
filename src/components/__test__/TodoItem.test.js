import { render, screen } from '@testing-library/react'
import { TodoItem } from './../TodoItem'

describe(`Footer component`, () => {
    const props = {
        id: 1,
        task: 'test',
        description: 'description',
        status: 'in-progress',
        created_on: '',
        last_updated_on: '',
        remove: jest.fn(),
        setStage: jest.fn()
    }
    it(`render`, () => {
        const { asFragment } = render(<TodoItem {...props} />)
        expect(asFragment()).toMatchSnapshot()
        const editButton = screen.getByText(/edit/i)
        const removeButton = screen.getByText(/x/i)
        editButton.click()
        removeButton.click()
    })
})