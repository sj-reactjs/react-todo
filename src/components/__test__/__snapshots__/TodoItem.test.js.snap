// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component render 1`] = `
<DocumentFragment>
  <div
    class="todo-card"
    role="listitem"
  >
    <div>
      <span
        data-action="edit"
        role="button"
        tabindex="0"
      >
        Edit
      </span>
      <span
        class="created"
      >
        N/A
      </span>
      <span
        class="right close"
        role="button"
        tabindex="0"
      >
        <span>
          x
        </span>
      </span>
    </div>
    <div
      class="title"
    >
      <h3>
        test
      </h3>
    </div>
    <div
      class="description"
    >
      <p>
        description
      </p>
    </div>
    <div
      class="status"
    >
      in-progress
    </div>
    <div
      class="updated"
    >
      N/A
    </div>
  </div>
</DocumentFragment>
`;
